internal/auth/auth.go

```diff
@@ -161,22 +180,48 @@ func (a *Authenticator) Wait(ctx context.Context, state State) (Result, error) {
 
 func (a *Authenticator) getDeviceCode(ctx context.Context) (State, error) {
 	data := url.Values{
-		"client_id": {a.ClientID},
-		"scope":     {strings.Join(requiredScopes, " ")},
-		"audience":  {a.Audience},
+		"client_id": []string{a.ClientID},
+		"scope":     []string{strings.Join(requiredScopes, " ")},
+		"audience":  []string{a.Audience},
 	}
-	r, err := http.PostForm(a.DeviceCodeEndpoint, data)
+
+	request, err := http.NewRequestWithContext(
```

Comment:
We're not using the http.PostForm because we want to keep using the passed in context.